### PR TITLE
Keep feature descriptions consistent with documentation

### PR DESCRIPTION
**Pull request by enjoy-binbin**
Recently we refined feature words, see apache/incubator-kvrocks-website#66.
This PR makes the key features in the README consistent with the docs,
plus removed some extra spaces and # characters.

### CODE REVIEW

**Comment by enjoy-binbin** [comment]
@PragmaTwice Hi, i suppose you can do the review, thanks.

File: README.md [hunk 1]
```diff
@@ -32,11 +32,11 @@
...
 
-* Redis protocol, user can use redis client to visit the kvrocks
-* Namespace, similar to redis db but use token per namespace
-* Replication, async replication using binlog like MySQL
-* High Available, supports redis sentinel to failover when master or slave was failed
-* Cluster mode, centralized management but compatible with Redis cluster client access
+* Redis Compatible: Users can access Apache Kvrocks via any Redis client.
+* Namespace: Similar to Redis SELECT but equipped with token per namespace.
+* Replication: Async replication using binlog like MySQL.
+* High Available: Support Redis sentinel to failover when master or slave was failed.
```
Comment (torwig) [inline README.md:38]: Here we can use `Highly Available` or `High Availability`.

Comment (enjoy-binbin) [reply README.md:38]: yean, my first thinking is that: want to be consistent with https://kvrocks.apache.org/
we can update the doc as well if needed

**Review by PragmaTwice — APPROVED** [review]
LGTM. Thanks for your contribution again.

**Review by git-hulk — APPROVED** [review]

**Pull request merged**